OHIF/Viewers
# fix(ViewportGrid) fill blank viewports with display sets not yet in grid.

In ViewportGrid, fill blank viewports with display sets not yet in grid.
ViewportGridService now allows off-screen viewports to remain so as to maintain continuity if they were filled by the UI/user.








### Context



When the layout is changed via the grid layout menu, the resulting viewports added to the layout should in general not be blank, but populated with display sets that have not been displayed in another viewport. This behaviour is consistent with OHIF v2. Furthermore, if once a viewport is populated with a display set, either by the user or by the auto-population introduced by this PR, it should maintain that display set.

### Changes & Results



An example of the (incorrect) behaviour prior to this PR, would be if the layout is changed from 1x1 to 2x2, the following would be displayed despite the fact there are several series/display sets that could/should be shown in the blank viewports.

![image](https://user-images.githubusercontent.com/109477394/217641713-a075035b-9903-4823-a604-b49b7ecbe712.png)

After the changes from this PR.

![image](https://user-images.githubusercontent.com/109477394/217642103-f698c35b-7bc4-46c1-963d-43a03626e2e6.png)

### Testing



1. Launch OHIF with a study with at least two series.
2. Switch layouts and note that any added viewports are populated by display sets not already present in a viewport on screen. Once all display sets have been displayed, then blank viewports are permissible.
3. Once a viewport is populated, it should maintain that display set - this includes if it is populated by dragging a display set into it.
4. Switching to MPR and back should also maintain populated viewports and/or auto-fill otherwise blank viewports.

### Checklist

#### PR



  semantic-release format and guidelines.

#### Code

  etc.)

#### Public Documentation Updates



  additions or removals.

#### Tested Environment

- [x] "Browser: Chrome 109.0.5414.120
  


[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch
